src/io/pfsisplug.c

```diff
@@ -66,6 +72,9 @@ void __osPfsRequestData(u8 cmd) {
     requestformat.dummy1 = CONT_CMD_NOP;
 
     for (i = 0; i < __osMaxControllers; i++) {
+        if ((__osControllerMask & (1 << i)) == 0) {
+            continue;
+        }
         *((__OSContRequesFormat*)ptr) = requestformat;
         ptr += sizeof(__OSContRequesFormat);
     }
```

Comment:
This looks like it's still broken, ptr isn't incremented if a channel is skipped